e2e/testcases/postsync_logging_test.go

```diff
@@ -320,37 +319,36 @@ func TestCloudLoggingStatusEventRepoSync(t *testing.T) {
 		nt.ClusterName,
 		startTime.Format(time.RFC3339),
 	)
-	nt.Must(waitForLogEntryInCloudLogs(nt, filter, 1, commit, repoSyncID.Name, repoSyncID.Namespace, []string{msg, "\"generation\":1"}, 120*time.Second))
+	nt.Must(waitForLogEntryInCloudLogs(nt, filter, commit, repoSyncID.Name, repoSyncID.Namespace, []string{msg, "\"generation\":1"}, 120*time.Second))
 }
 
-// assertLogEntryHasCount checks if the specified messages appear in the logs
-// exactly the expected number of times, matching the given commit, name, and namespace.
-func assertLogEntryHasCount(logs []string, expectedCount int, commit, rname, rnamespace string, messages []string) error {
-	count := 0
-
+// assertOneLogLinePerMessage checks if each specified message appears in exactly one line
+// among the log entries which match the given commit, resource name, and namespace.
+func assertOneLogLinePerMessage(logs []string, commit, resourceName, resourceNamespace string, messages []string) error {
+	messageCounts := make(map[string]int)
+	for _, msg := range messages {
+		messageCounts[msg] = 0
+	}
 	for _, line := range logs {
-		if strings.Contains(line, commit) && strings.Contains(line, rname) && strings.Contains(line, rnamespace) {
-			allFound := false
-			for _, msg := range messages {
-				if !strings.Contains(line, msg) {
-					break
-				}
-				allFound = true
-			}
-			if allFound {
-				count++
+		if !(strings.Contains(line, commit) && strings.Contains(line, resourceName) && strings.Contains(line, resourceNamespace)) {
+			continue
+		}
+		for _, msg := range messages {
+			if strings.Contains(line, msg) {
+				messageCounts[msg]++
 			}
 		}
 	}
-	if count != expectedCount {
-		return fmt.Errorf("Expected %d occurrences of error log entry %s, but found %d",
-			expectedCount, messages, count)
+	for _, msg := range messages {
+		if count, ok := messageCounts[msg]; !ok || count != 1 {
+			return fmt.Errorf("failed to find single occurrence of each message in logs, counted:\n%v\nin logs:\n%v", messageCounts, logs)
```

Comment:
How many logs are typically in this array? I'm concerned this error may be overly verbose, granted it would probably be useful for debugging

Reply:
Probably not necessary, removed